content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx

```diff
@@ -0,0 +1,186 @@
+---
+title: 'Names in the underlying database'
+metaDescription: ''
+tocDepth: 2
+---
+
+<TopBlock>
+
+The [Prisma schema](./) includes mechanisms that allow you to define names of certain database objects. You can:
+
+- [Map model and field names to different collection/table and field/column names in the underlying database](#mapping-collectiontable-and-fieldcolumn-names)
+- [Define constraint and index names](#constraint-and-index-names-preview) (Preview)
+
+</TopBlock>
+
+## Mapping collection/table and field/column names
+
+Prisma model [naming conventions (singular form, PascalCase)](../../../reference/api-reference/prisma-schema-reference#naming-conventions) <span class="api"></span> do not always match table names in the database. A common approach for naming tables/collections in databases is to use plural form and [snake_case](https://en.wikipedia.org/wiki/Snake_case) notation.
+
+For example, when you introspect a database with a table named `comments`, the result Prisma model will look like this:
+
+```prisma
+model comments {
+   // Fields
+}
+```
+
+However, you can still follow the naming convention without renaming the underlying `comments` table in the database by using the [`@@map`](../../../reference/api-reference/prisma-schema-reference#map-1) <span class="api"></span> attribute:
+
+```prisma
+model Comment {
+   // Fields
+  @@map("comments")
+}
+```
+
+With this modified model definition, Prisma automatically maps the `Comment` model to the `comments` table in the underlying database.
+
+You can also [`@map`](../../../reference/api-reference/prisma-schema-reference#map) <span class="api"></span> a column/field name:
+
+```prisma highlight=2-4;normal
+model Comment {
+  content String  @map("comment_text")
+  email   String  @map("commenter_email")
+  type    Enum    @map("comment_type")
+
+  @@map("comments")
+}
+```
+
+You can also `@map` an enum value, or `@@map` an enum:
+
+```prisma highlight=3,5;normal
+enum Type {
+  Blog,
+  Twitter @map("comment_twitter")
+
+  @@map("comment_source_enum")
+}
+```
+
+`@map` and `@@map` allow you to [tune the shape of your Prisma Client API](../prisma-client/working-with-prismaclient/use-custom-model-and-field-names#using-map-and-map-to-rename-fields-and-models-in-the-prisma-client-api) by decoupling model and field names from table and column names in the underlying database.
+
+## Constraint and index names (Preview)
```

Comment:
I think this whole section could use a mention of what happens when you introspect or migrate a schema that uses or doesn't use this. It is also unclear where these constraint and index names actually _affect_ anything in the Client! A code example would probably be nice to show how this can make the API to be used for a query nicer.

Reply:
put in the content, still ugly though.